File: java_doc/Portfolio.java
/**
 * Predstavitveni razred za osebni portfelj.
 * Vsebuje osnovne podatke (ime, e-pošta, GitHub) in metode za izpis predstavitve.
 *
 * <p>Namen: prikaz uporabe Javadoc v okviru naloge RPO.</p>
 *
 * @author Anastasija Temova
 * @version 1.0
 */
public class Portfolio {

    private String authorName;
    private String email;
    private String githubLink;

    /**
     * Ustvari nov portfelj.
     *
     * @param authorName ime avtorja
     * @param email e-pošta avtorja
     * @param githubLink povezava do GitHub profila
     */
    public Portfolio(String authorName, String email, String githubLink) {
        this.authorName = authorName;
        this.email = email;
        this.githubLink = githubLink;
    }

    /**
     * Nastavi e-pošto (preveri osnovno veljavnost).
     *
     * @param email nova e-pošta
     * @throws IllegalArgumentException če je e-pošta prazna ali brez znaka '@'
     */
    public void setEmail(String email) {
        if (email == null || email.isBlank() || !email.contains("@")) {
            throw new IllegalArgumentException("Neveljaven e-poštni naslov.");
        }
        this.email = email;
    }

    /**
     * Vrne kratko predstavitev avtorja.
     *
     * @return niz s kratko predstavitvijo
     */
    public String getIntroduction() {
        return "Hello, I am " + authorName + " — computer science student. "
                + "You can write to me " + email + " or see my work here: " + githubLink + ".";
    }

    /**
     * Izpiše predstavitev v konzolo.
     */
    public void printIntroduction() {
        System.out.println(getIntroduction());
    }

    /**
     * Vstopna točka programa (demo).
     *
     * @param args argumenti ukazne vrstice (neuporabljeno)
     */
    public static void main(String[] args) {
        Portfolio p = new Portfolio(
                "Anastasija Temova",
                "temanastasa@gmail.com",
                "https://github.com/ATemova"
        );
        p.printIntroduction();
    }
}

File: python_src/portfolio.py
"""
Ta modul predstavlja osnovne podatke o portfoliu in izpiše
kratko predstavitev avtorja.
"""

from dataclasses import dataclass


@dataclass
class Portfolio:
    """Predstavlja preprost osebni portfolio.

    Atributi:
        author_name: Ime in priimek lastnika portfolia.
        email: Kontaktni e-naslov.
        github: Povezava do GitHub profila.

    Primer:
        >>> p = Portfolio("Anastasija Temova", "temanastasa@gmail.com", "https://github.com/ATemova")
        >>> p.intro()
        'Hello, I am Anastasija Temova. Contact: temanastasa@gmail.com • GitHub: https://github.com/ATemova'
    """

    author_name: str
    email: str
    github: str

    def intro(self) -> str:
        """Vrne kratek opis portfolia.

        Vrne:
            Niz z osnovnimi informacijami o avtorju.
        """
        return f"Hello, I am {self.author_name}. Contact: {self.email} • GitHub: {self.github}"


def preveri_email(email: str) -> bool:
    """Preveri, ali e-naslov vsebuje znak '@'.

    Argumenti:
        email: E-naslov, ki ga želimo preveriti.

    Vrne:
        True, če e-naslov vsebuje '@', sicer False.
    """
    return isinstance(email, str) and "@" in email